google-cloud-server/src/main/kotlin/jetbrains/buildServer/clouds/google/connector/GoogleApiConnectorImpl.kt

```diff
@@ -119,14 +119,24 @@ class GoogleApiConnectorImpl : GoogleApiConnector {
             }
         }
 
+
+        @Suppress("IMPLICIT_CAST_TO_ANY")
+        val instanceBootImage = when(details.imageType) {
+            GoogleCloudImageType.Image -> ProjectGlobalImageName.format(details.sourceImage, myProjectId)
+            GoogleCloudImageType.ImageFamily -> ProjectGlobalImageFamilyName.format(details.sourceImageFamily, myProjectId)
+            else -> LOG.warn("Invalid imageType: ${details.imageType}")
+        }
+
+        LOG.info("Creating instance from ${details.imageType}, using source: $instanceBootImage")
+
         val instanceInfo = getInstanceBuilder(instance)
-                .setMachineType(ProjectZoneMachineTypeName.of(machineType, myProjectId, zone).value)
+                .setMachineType(ProjectZoneMachineTypeName.format(machineType, myProjectId, zone))
```

Comment:
Judging by the codebase, this change introduces [the regression](https://github.com/JetBrains/teamcity-google-agent/issues/7) that was fixed by the [commit](https://github.com/JetBrains/teamcity-google-agent/commit/391ac4802943b860f56230f1ef72fd29315e9798). Could you please, change all the `Project<...>.of` -> `Project<...>.format` changes back to `Project<...>.of`?

Reply:
So we've been running a snapshot build with these changes since opening this PR, and haven't had any issues or errors such as those in the linked regression...

So maybe bumping the SDK version has fixed that error? 

I'll run some more tests though...

Reply:
So I had another look at the Google compute SDK, and from what I can see the `.format` call does basically the same as `.of().value()`.

E.g. `ProjectZoneMachineTypeNme` calls https://github.com/googleapis/java-compute/blob/f7bfd83ba5f4c0037f1ffb712fe56d8675f1cbf5/src/main/java/com/google/cloud/compute/v1/ProjectZoneMachineTypeName.java#L60-L64, which ultimately calls https://github.com/googleapis/java-compute/blob/f7bfd83ba5f4c0037f1ffb712fe56d8675f1cbf5/src/main/java/com/google/cloud/compute/v1/ProjectZoneMachineTypeName.java#L169-L175

Reply:
Hmm, I looked into an old version of SDK, probably. Thanks!